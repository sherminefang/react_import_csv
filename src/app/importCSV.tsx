import 'bootstrap/dist/css/bootstrap.min.css';
import "react-bootstrap-table-next/dist/react-bootstrap-table2.min.css";
import React, {useState} from "react";
import BootstrapTable from "react-bootstrap-table-next";
import Button from "react-bootstrap/Button";
import paginationFactory from "react-bootstrap-table2-paginator";
import filterFactory, { textFilter } from 'react-bootstrap-table2-filter';
import ToolkitProvider, { Search } from 'react-bootstrap-table2-toolkit';
import { useCSVReader } from 'react-papaparse';

export function ImportCSV() {
  const [array, setArray] = useState([]);
  const { CSVReader } = useCSVReader();
  const headerKeys = Object.keys(Object.assign({}, ...array));

  const { SearchBar } = Search;
  const columns = [
    {
      dataField: "postId",
      text: "Post ID",
    },
    {
      dataField: "id",
      text: "ID",
      sort: true
    },
    {
      dataField: "name",
      text: "Name",
      sort: true
    },
    {
      dataField: "email",
      text: "Email",
      sort: true
    },
    {
      dataField: "body",
      text: "Body"
    }
  ];

  return (
      <>
        <h1 style={{textAlign: "center", paddingBlock: "1rem"}}>CSV IMPORT</h1>
          <CSVReader
            onUploadAccepted={(results:any) => {
              setArray(results.data);
            }}
            config={{
              header: true,
              dynamicTyping: true,
              skipEmptyLines: true,
            }}>
          {({
            getRootProps,
            acceptedFile,
            ProgressBar
          }: any) => (
            <>
              <Button  {...getRootProps()} style={{marginBlockEnd:"1rem"}} >
                  Browse file
                </Button>
                  {acceptedFile && acceptedFile.name}
                  <ProgressBar animated />
                  </>
                )}
          </CSVReader>     
          <ToolkitProvider
            keyField="id"
            data={ array }
            columns={ columns }
            search>
            {
              props => (
                <div>
                  <SearchBar { ...props.searchProps } />
                  <hr />
                  <BootstrapTable
                  bootstrap4
                  filter={ filterFactory() }
                  pagination={paginationFactory({ sizePerPage: 5 })}
                  { ...props.baseProps }
                />
                </div>
              )
            }
          </ToolkitProvider>
      </>
  );
}
