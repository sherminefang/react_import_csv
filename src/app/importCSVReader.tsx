import 'bootstrap/dist/css/bootstrap.min.css';
import "react-bootstrap-table-next/dist/react-bootstrap-table2.min.css";
import React, {useState} from "react";
import BootstrapTable from "react-bootstrap-table-next";
import Button from "react-bootstrap/Button";
import paginationFactory from "react-bootstrap-table2-paginator";
import filterFactory from 'react-bootstrap-table2-filter';
import ToolkitProvider, { Search } from 'react-bootstrap-table2-toolkit';
import { useCSVReader } from 'react-papaparse';

interface ImportCSVReaderProps {
  onUploadAccepted?: ((data: any, file?: any) => void) | undefined;
}

export const ImportCSVReader: React.FC<ImportCSVReaderProps> = (
  props: ImportCSVReaderProps
) => {
  const { CSVReader } = useCSVReader();
  const { onUploadAccepted } = props;
  const [array, setArray] = useState([]);

  const { SearchBar } = Search;
  const columns = [
    {
      dataField: "postId",
      text: "Post ID",
    },
    {
      dataField: "id",
      text: "ID",
      sort: true
    },
    {
      dataField: "name",
      text: "Name",
      sort: true
    },
    {
      dataField: "email",
      text: "Email",
      sort: true
    },
    {
      dataField: "body",
      text: "Body"
    }
  ];

  return (
    <>
    <CSVReader
            onUploadAccepted={(results:any) => {
              setArray(results.data);
            }}
            accept='text/csv, .csv'
            config={{
              header: true,
              dynamicTyping: true,
              skipEmptyLines: true,
            }}>
          {({
            getRootProps,
            acceptedFile,
            ProgressBar
          }: any) => (
            <>
              <Button  {...getRootProps()} style={{marginBlockEnd:"1rem"}} >
                  Browse file
                </Button>
                  {acceptedFile && acceptedFile.name}
                  <ProgressBar animated />
                  </>
                )}
          </CSVReader>     
          <ToolkitProvider
            keyField="id"
            data={ array }
            columns={ columns }
            search>
            {
              props => (
                <div>
                  <SearchBar { ...props.searchProps } />
                  <hr />
                  <BootstrapTable
                  bootstrap4
                  filter={ filterFactory() }
                  pagination={paginationFactory({ sizePerPage: 5 })}
                  { ...props.baseProps }
                />
                </div>
              )
            }
          </ToolkitProvider>
    </>
  );
};

