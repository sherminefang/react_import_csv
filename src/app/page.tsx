"use client";
//import { ImportCSV} from "./importCSV"
import { ImportCSVReader } from "./importCSVReader";
export default function Home() {
  return (
    <div className="container" style={{paddingBlockStart: "1rem", paddingBlockEnd: "200px"}}>
        <h1 style={{textAlign: "center", paddingBlock: "1rem"}}>CSV IMPORT</h1>
       {/* <ImportCSV /> */}
       <ImportCSVReader />
    </div>
  )  
}
