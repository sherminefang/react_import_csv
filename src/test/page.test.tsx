import React from 'react';
import { act, screen, render } from '@testing-library/react'
import {ImportCSVReader} from '../app/importCSVReader';
import Home from '../app/page';

describe('Render tests', () => {
    it('render correctly', () => {
        render(<Home/>);
        const headingElem = screen.getByRole('heading', { level: 1 });
        expect(headingElem).toBeInTheDocument();
    });

    it('render button correctly', () => {
        render(<Home/>);
        const buttonElem = screen.getByText('Browse file');
        expect(buttonElem).toBeInTheDocument();
    });

    it('render table correctly', () => {
        render(<Home/>);
        const tableElem = screen.getByRole('table');
        expect(tableElem).toBeInTheDocument();
    });

    it('render search correctly', () => {
        render(<Home/>);
        const searchElem = screen.getByPlaceholderText('Search');
        expect(searchElem).toBeInTheDocument();
    });

    it('render pagination correctly', () => {
        render(<Home/>);
        const paginationElem = screen.getByRole('menu');
        expect(paginationElem).toBeInTheDocument();
    });
});

describe('Render tests', () => {
    it('render correctly', () => {
        const handleUploadAccepted = jest.fn().mockImplementation();
        const tree = render(<ImportCSVReader onUploadAccepted={handleUploadAccepted} />)
        expect(tree).toMatchSnapshot();
    });
});



// describe('Component tests', () => {
//     const handleUploadAccepted = jest.fn().mockImplementation();
//     const tree = render(<ImportCSVReader onUploadAccepted={handleUploadAccepted}/>)

//     // it('should contains the button', () => {
//     //     expect(tree('btn')).toBeInTheDocument();
//     // });

//     it("should have", () => {
//         expect(tree).toMatchSnapshot();
//     });
// });



// import Home from '../app/page';

// test('renders learn react link', () => {
//     render(<Home />);

//     const linkElement = screen.getByText(/learn react/i);
  
//     expect(linkElement).toBeInTheDocument();
  
//   });

// // // eslint-disable-next-line no-undef
// // describe('CSVReader', () => {
// //   // eslint-disable-next-line no-undef
// //   it('able extract file', async function () {
// //     const handleUploadAccepted = jest.fn().mockImplementation();
// //     const tree = renderer.create(
// //         <ImportCSVReader
// //           onUploadAccepted={handleUploadAccepted} />
// //       )
// //       .toJSON();
// //     expect(tree).toMatchSnapshot();
// //   });
// // });

