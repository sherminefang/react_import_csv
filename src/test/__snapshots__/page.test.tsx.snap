// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render tests render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        accept="text/csv, .csv"
        autocomplete="off"
        style="display: none;"
        tabindex="-1"
        type="file"
      />
      <button
        class="btn btn-primary"
        data-testid="btn"
        style="margin-block-end: 1rem;"
        type="button"
      >
        Browse file
      </button>
      <span
        style="border-radius: 3px; box-shadow: inset 0 1px 3px rgba(0, 0, 0, .2); bottom: 14px; width: 0%; background-color: rgb(101, 156, 239); height: 10px; transition: width 500ms ease-in-out; display: none;"
      />
      <div>
        <label
          class="search-label"
          for="search-bar-0"
        >
          <span
            class="sr-only"
            id="search-bar-0-label"
          >
            Search this table
          </span>
          <input
            aria-labelledby="search-bar-0-label"
            class="form-control "
            id="search-bar-0"
            placeholder="Search"
            type="text"
            value=""
          />
        </label>
        <hr />
        <div
          class="react-bootstrap-table"
        >
          <table
            class="table table-bordered"
          >
            <thead>
              <tr>
                <th
                  tabindex="0"
                >
                  Post ID
                </th>
                <th
                  aria-label="ID sortable"
                  class="sortable"
                  tabindex="0"
                >
                  ID
                  <span
                    class="order"
                  >
                    <span
                      class="dropdown"
                    >
                      <span
                        class="caret"
                      />
                    </span>
                    <span
                      class="dropup"
                    >
                      <span
                        class="caret"
                      />
                    </span>
                  </span>
                </th>
                <th
                  aria-label="Name sortable"
                  class="sortable"
                  tabindex="0"
                >
                  Name
                  <span
                    class="order"
                  >
                    <span
                      class="dropdown"
                    >
                      <span
                        class="caret"
                      />
                    </span>
                    <span
                      class="dropup"
                    >
                      <span
                        class="caret"
                      />
                    </span>
                  </span>
                </th>
                <th
                  aria-label="Email sortable"
                  class="sortable"
                  tabindex="0"
                >
                  Email
                  <span
                    class="order"
                  >
                    <span
                      class="dropdown"
                    >
                      <span
                        class="caret"
                      />
                    </span>
                    <span
                      class="dropup"
                    >
                      <span
                        class="caret"
                      />
                    </span>
                  </span>
                </th>
                <th
                  tabindex="0"
                >
                  Body
                </th>
              </tr>
            </thead>
          </table>
        </div>
        <div
          class="row react-bootstrap-table-pagination"
        >
          <div
            class="col-md-6 col-xs-6 col-sm-6 col-lg-6"
          >
            <span
              class="react-bs-table-sizePerPage-dropdown dropdown"
              style="visibility: visible;"
            >
              <button
                aria-expanded="false"
                class="btn btn-default btn-secondary dropdown-toggle"
                data-toggle="dropdown"
                id="pageDropDown"
                type="button"
              >
                5
                 
                <span>
                  <span
                    class="caret"
                  />
                </span>
              </button>
              <ul
                aria-labelledby="pageDropDown"
                class="dropdown-menu "
                role="menu"
              >
                <li
                  class="dropdown-item"
                  role="presentation"
                >
                  <a
                    data-page="10"
                    href="#"
                    role="menuitem"
                    tabindex="-1"
                  >
                    10
                  </a>
                </li>
                <li
                  class="dropdown-item"
                  role="presentation"
                >
                  <a
                    data-page="25"
                    href="#"
                    role="menuitem"
                    tabindex="-1"
                  >
                    25
                  </a>
                </li>
                <li
                  class="dropdown-item"
                  role="presentation"
                >
                  <a
                    data-page="30"
                    href="#"
                    role="menuitem"
                    tabindex="-1"
                  >
                    30
                  </a>
                </li>
                <li
                  class="dropdown-item"
                  role="presentation"
                >
                  <a
                    data-page="50"
                    href="#"
                    role="menuitem"
                    tabindex="-1"
                  >
                    50
                  </a>
                </li>
              </ul>
            </span>
          </div>
          <div
            class="react-bootstrap-table-pagination-list col-md-6 col-xs-6 col-sm-6 col-lg-6"
          >
            <ul
              class="pagination react-bootstrap-table-page-btns-ul"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <input
      accept="text/csv, .csv"
      autocomplete="off"
      style="display: none;"
      tabindex="-1"
      type="file"
    />
    <button
      class="btn btn-primary"
      data-testid="btn"
      style="margin-block-end: 1rem;"
      type="button"
    >
      Browse file
    </button>
    <span
      style="border-radius: 3px; box-shadow: inset 0 1px 3px rgba(0, 0, 0, .2); bottom: 14px; width: 0%; background-color: rgb(101, 156, 239); height: 10px; transition: width 500ms ease-in-out; display: none;"
    />
    <div>
      <label
        class="search-label"
        for="search-bar-0"
      >
        <span
          class="sr-only"
          id="search-bar-0-label"
        >
          Search this table
        </span>
        <input
          aria-labelledby="search-bar-0-label"
          class="form-control "
          id="search-bar-0"
          placeholder="Search"
          type="text"
          value=""
        />
      </label>
      <hr />
      <div
        class="react-bootstrap-table"
      >
        <table
          class="table table-bordered"
        >
          <thead>
            <tr>
              <th
                tabindex="0"
              >
                Post ID
              </th>
              <th
                aria-label="ID sortable"
                class="sortable"
                tabindex="0"
              >
                ID
                <span
                  class="order"
                >
                  <span
                    class="dropdown"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                  <span
                    class="dropup"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                </span>
              </th>
              <th
                aria-label="Name sortable"
                class="sortable"
                tabindex="0"
              >
                Name
                <span
                  class="order"
                >
                  <span
                    class="dropdown"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                  <span
                    class="dropup"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                </span>
              </th>
              <th
                aria-label="Email sortable"
                class="sortable"
                tabindex="0"
              >
                Email
                <span
                  class="order"
                >
                  <span
                    class="dropdown"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                  <span
                    class="dropup"
                  >
                    <span
                      class="caret"
                    />
                  </span>
                </span>
              </th>
              <th
                tabindex="0"
              >
                Body
              </th>
            </tr>
          </thead>
        </table>
      </div>
      <div
        class="row react-bootstrap-table-pagination"
      >
        <div
          class="col-md-6 col-xs-6 col-sm-6 col-lg-6"
        >
          <span
            class="react-bs-table-sizePerPage-dropdown dropdown"
            style="visibility: visible;"
          >
            <button
              aria-expanded="false"
              class="btn btn-default btn-secondary dropdown-toggle"
              data-toggle="dropdown"
              id="pageDropDown"
              type="button"
            >
              5
               
              <span>
                <span
                  class="caret"
                />
              </span>
            </button>
            <ul
              aria-labelledby="pageDropDown"
              class="dropdown-menu "
              role="menu"
            >
              <li
                class="dropdown-item"
                role="presentation"
              >
                <a
                  data-page="10"
                  href="#"
                  role="menuitem"
                  tabindex="-1"
                >
                  10
                </a>
              </li>
              <li
                class="dropdown-item"
                role="presentation"
              >
                <a
                  data-page="25"
                  href="#"
                  role="menuitem"
                  tabindex="-1"
                >
                  25
                </a>
              </li>
              <li
                class="dropdown-item"
                role="presentation"
              >
                <a
                  data-page="30"
                  href="#"
                  role="menuitem"
                  tabindex="-1"
                >
                  30
                </a>
              </li>
              <li
                class="dropdown-item"
                role="presentation"
              >
                <a
                  data-page="50"
                  href="#"
                  role="menuitem"
                  tabindex="-1"
                >
                  50
                </a>
              </li>
            </ul>
          </span>
        </div>
        <div
          class="react-bootstrap-table-pagination-list col-md-6 col-xs-6 col-sm-6 col-lg-6"
        >
          <ul
            class="pagination react-bootstrap-table-page-btns-ul"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
